src/cython/cython/cubical_complex.pyx

```diff
@@ -195,4 +197,4 @@ cdef class CubicalComplex:
         else:
             print("intervals_in_dim function requires persistence function"
                   " to be launched first.")
-        return intervals_result
+        return np_array(intervals_result)
```

Comment:
Does that mean that cython creates a list of pairs, and we convert it to a numpy array afterwards? That's ok as a starting point to define the interface, but eventually we will want to produce a numpy array directly.

Reply:
(Hmm github seems to have lost my comment, probably I clicked on "resolve" and thought it would also post the text) The intermediate object we convert to a numpy array is not a python list, it is a C++ish vector of pairs (via cython), hopefully converting that to a numpy array is not as slow as building an intermediate list would be. Really returning a numpy array from C++ code seems surprisingly hard from what I have seen (and not just with cython), so I think this will do for now.